### PR TITLE
Add stream after transfer benchmark.

### PR DESCRIPTION
**Pull request by MichalMrozek**
Measures impact of transfer operation preceding stream kernel.

Signed-off-by: Michal Mrozek <michal.mrozek@intel.com>

### CODE REVIEW

File: TESTS.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ ReadBufferRect|allocates an OpenCL buffer and measures rectangle read bandwidth.
...
 SLM_DataAccessLatency|generates SLM local memory transactions inside thread group to measure latency between reads (uses Intel only private intel_get_cycle_counter() )|<ul><li>--direction write or read mode (0 or 1)</li><li>--occupancyDiv H/W load divider by 8, 4, 2, full occupancy</li><li>--size SLM Size</li></ul>|:x:|:heavy_check_mark:|
+StreamAfterTransfer|Goal of this test is to meassure how stream kernels perform right after host to device transfer populating the data. Test does clean caches, then emits transfers and then follows with stream kernel and meassures GPU execution time of it.|<ul><li>--size Size of the memory to stream. Must be divisible by datatype size.</li><li>--type Memory streaming type (Read or Write or Scale or Triad)</li><li>--useEvents Perform GPU-side measurements using events (0 or 1)</li></ul>|:x:|:heavy_check_mark:|
```
Comment (jandres742) [inline TESTS.md:83]: measure is misspelled

File: source/benchmarks/memory_benchmark/implementations/ocl/stream_after_transfer_ocl.cpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            std::cout << buffer.get() << std::endl;
+        }
+        return result;
+    }
+    cl_kernel kernel = clCreateKernel(program, kernelName, &retVal);
+    ASSERT_CL_SUCCESS(retVal);
+    cl_kernel cacheCleaner = clCreateKernel(program, "write", &retVal);
+    ASSERT_CL_SUCCESS(retVal);
+
+    auto cleanerSize = 1 * 1024 * 1024 * 1024u; // 1GB
```
Comment (jandres742) [inline source/benchmarks/memory_benchmark/implementations/ocl/stream_after_transfer_ocl.cpp:89]: just 1073741824 ?

Comment (MichalMrozek) [reply source/benchmarks/memory_benchmark/implementations/ocl/stream_after_transfer_ocl.cpp]: changed

File: TESTS.md (hunk 1 above)
Comment (MichalMrozek) [reply TESTS.md:83]: fixed

**Pull request closed without merging**